src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -665,9 +662,13 @@ internal bool IsAutoProperty
         protected bool AccessorsHaveImplementation
             => (_propertyFlags & Flags.AccessorsHaveImplementation) != 0;
 
+        internal bool HasSynthesizedBackingField
```

Comment:
I've reverted the name to `IsAutoPropertyOrUsesFieldKeyword` for now. Ideally, this property should be renamed to `IsAutoProperty` and replace the current `IsAutoProperty`. But the name `HasSynthesizedBackingField` or `HasBackingField` is misleading because it's not equivalent to `BackingField is { }` since `BackingField` is set in the error case where a non-auto-property includes an initializer.